File: library/src/main/java/me/proxer/library/api/DelimitedStringSet.java
package me.proxer.library.api;

import com.squareup.moshi.JsonQualifier;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @author Ruben Gees
 */
@Retention(RetentionPolicy.RUNTIME)
@JsonQualifier
public @interface DelimitedStringSet {

    String delimiter() default " ";

    String[] valuesToKeep() default {};
}

File: library/src/main/kotlin/me/proxer/library/internal/adapter/DelimitedStringSet.kt
package me.proxer.library.internal.adapter

import com.squareup.moshi.JsonQualifier

/**
 * @author Ruben Gees
 */
@JsonQualifier
@Retention(AnnotationRetention.RUNTIME)
internal annotation class DelimitedStringSet(
    val delimiter: String = " ",
    val valuesToKeep: Array<String> = []
)
